plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/CameraUpdateProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdate;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+class CameraUpdateProvider implements RefreshPluginRunnable.RefreshPluginListener {
+
+  private final MapboxMap mapboxMap;
+  private final OnCameraMoveInvalidateListener onCameraMoveInvalidateListener;
+  private final RefreshPluginRunnable refreshPluginRunnable;
+  private float bearing = -1f;
+  private LatLng target;
+
+  CameraUpdateProvider(MapboxMap mapboxMap,
+                       OnCameraMoveInvalidateListener onCameraMoveInvalidateListener,
+                       long refreshIntervalMillis) {
+    this.mapboxMap = mapboxMap;
+    this.onCameraMoveInvalidateListener = onCameraMoveInvalidateListener;
+    refreshPluginRunnable = new RefreshPluginRunnable(refreshIntervalMillis, this);
+    refreshPluginRunnable.run();
+  }
+
+  @Override
+  public void onShouldRefresh() {
+    CameraUpdate cameraUpdate = buildCameraUpdate();
+    mapboxMap.moveCamera(cameraUpdate);
+    onCameraMoveInvalidateListener.onInvalidateCameraMove();
+  }
+
+  void updateRefreshInterval(long refreshIntervalMillis) {
+    refreshPluginRunnable.updateRefreshInterval(refreshIntervalMillis);
+  }
+
+  void pushBearing(float bearing) {
+    this.bearing = bearing;
+  }
+
+  void pushLatLng(LatLng target) {
+    this.target = target;
+  }
+
+  private CameraUpdate buildCameraUpdate() {
+    CameraPosition cameraPosition = mapboxMap.getCameraPosition();
+    CameraPosition.Builder cameraPositionBuilder = new CameraPosition.Builder();
+    if (bearing > 0 && bearing != cameraPosition.bearing) {
```

Comment:
I think our current logic doesn't ensure that the target bearing is always 0-360.

Reply:
Does the shortest rotation util address this? https://github.com/mapbox/mapbox-plugins-android/blob/master/plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/Utils.java#L32-L40

Reply:
I don't think that's the case if the previous and current bearings are >360 or <-360, and when both are < 0. We probably should update the utils method to better normalize the bearing. Also, the bearing can be equal to 0.

Reply:
Additionally, we need to check here if we actually want to modify camera's bearing, otherwise in a mode like `CameraModer.TRACKING`, where we are tracking only target, we are unable to rotate the camera.